src/content/dates.test.ts

```diff
@@ -0,0 +1,109 @@
+import { describe, expect, it } from 'vitest';
+
+import {
+  EraInfoDate,
+  getEraInfoTimeSpan,
+  parseEraDate,
+  startsWithEraName,
+} from './dates';
+
+describe('era date conversion', () => {
```

Comment:
Just a curiosity as much as anything: from what I understand we deliberately avoid tests that require converting pre-Gregorian calendar dates since that requires calling the background script and asynchronously loading the browser polyfill too.

Would it make sense to mock `'webextension-polyfill'` to cover just the sendMessage function and have it call the code in `calculate-date.ts` here?

I believe we mock `webextension-polyfill` in `jpdict.test.ts` and `commands.test.ts`. If that works for us here I think it would let us:

- Put all our tests in the one file
- Avoid doing the async import in `dates.ts`

It would make the tests more like integration tests than unit tests but I think that's preferable.

Reply:
> from what I understand we deliberately avoid tests that require converting pre-Gregorian calendar dates since that requires calling the background script and asynchronously loading the browser polyfill too.

Yes, exactly.

> Would it make sense to mock `'webextension-polyfill'` to cover just the sendMessage function and have it call the code in `calculate-date.ts` here?

Sure, this seems like a good change to make. I will implement this, thanks for the suggestion!

Reply:
Looks like we have problems in the mocha tests in `get-text.test.ts` now.

Reply:
Oh, that's a shame. I wonder if we can mock the `webextension-polyfill` there too.

Reply:
I will look into this.
You may want to cancel the currently running GitHub Actions tests manually, as they seem to be running indefinitely.

Reply:
Thanks! I've cancelled those actions now. I started looking into `playwright-test` but I haven't yet found out a way to mock that import.

Reply:
The closest I've found is that it looks like you can customize the build options passed to esbuild from `playwright-test` so perhaps we can use the [alias](https://esbuild.github.io/api/#alias) field to alias `webextension-playwright` to a mock version that just implements the functions we need for browser-based testing.

Reply:
It works with your proposed solution! And I found a way to get Cosmos working along the way (I will submit a seperate PR for this later). Also I restructured the EraInfoComponent to fetch the time span data internally, as Cosmos can also use the browser-polyfill mock.

I did minimal changes to the `tests` folder, though. I'm not really sure, how everything is structured. Do some files need to be compiled manually there?

I'm keeping the commit history for the time being, so it's easier for you to track the changes I made. I will tidy it up later.

Reply:
> It works with your proposed solution! And I found a way to get Cosmos working along the way (I will submit a seperate PR for this later). Also I restructured the EraInfoComponent to fetch the time span data internally, as Cosmos can also use the browser-polyfill mock.

Wow, amazing work!

> I did minimal changes to the `tests` folder, though. I'm not really sure, how everything is structured. Do some files need to be compiled manually there?

Unfortunately that folder is a bit of a mess. After we convert everything to Preact we should be able to remove some of the mess there, I hope.

I believe rspack builds `content-loader.ts` as a separate (parallel) project but I hope we can ultimately get rid of that.

I don't _think_ there's anything that needs to be compiled manually.

> I'm keeping the commit history for the time being, so it's easier for you to track the changes I made. I will tidy it up later.

Thank you! I've reviewed the changes now and they look great so feel free to tidy up the commits any time!

Reply:
> I don't _think_ there's anything that needs to be compiled manually.

Hmm okay, I was wondering because `content-loader.js` didn't change after I made some changes to `content-loader.ts`.